### ISSUE TITLE
issue in print_top_misclassified

### ISSUE THREAD

**Issue by ddabhane** [iit-cs579/main#311]
predict_proba returns a list of list where each list has 2 values [n_samples, n_classes].

**When I am sorting in descending order of n_samples i get the following output of the one's required from log.text:**
truth=1 predicted=0 proba=0.990791
In defense of this movie I must repeat what I had stated previously......

truth=1 predicted=0 proba=0.990739
Being a freshman in college, this movie reminded me of my.......

truth=1 predicted=0 proba=0.974652
Ah, classic comedy. At the point in the movie where brains get .......

**When I am sorting in descending order of n_classes I get the following output:**

truth=0 predicted=1 proba=0.993731
I absolutely despise this film. I wanted to love it .....

truth=0 predicted=1 proba=0.991584
When I attended college in the early 70s, i.......

**How can I fix this?**

**Comment by AndrewLu1992** [comment]
Quote from sklearn documentation:

"""
predict_proba(X)[source]
Probability estimates.
The returned estimates for all classes are ordered by the label of classes.
For a multi_class problem, if multi_class is set to be “multinomial” the softmax function is used to find the predicted probability of each class. Else use a one-vs-rest approach, i.e calculate the probability of each class assuming it to be positive using the logistic function. and normalize these values across all the classes.
Parameters:	
X : array-like, shape = [n_samples, n_features]
Returns:	
T : array-like, shape = [n_samples, n_classes]
Returns the probability of the sample for each class in the model, where classes are ordered as they are in self.classes_.
"""

This function returns a 2darray where each row corresponds to a document and values are the probability that this document should be labeled to each class.

It is not predicts by prob[:, 0] or prob[0, :].

Thus you need to rank by the probability corresponding to what the trained classifier predicts. If it predicts class 1, then take prob associated with class 1 for that document, If predicting class 2, then take associated prob and etc...
